streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -193,6 +195,22 @@ public int hashCode() {
     public TopologyBuilder() {}
 
     /**
+     * Set the applicationId to be used for auto-generated internal topics.
+     *
+     * This is required before calling {@link #sourceTopics}, {@link #topicGroups},
+     * {@link #copartitionSources} and {@link #build(Integer)}.
+     *
+     * @param applicationId the streams applicationId. Should be the same as set by
+     * {@link org.apache.kafka.streams.StreamsConfig#APPLICATION_ID_CONFIG}
+     */
+    public synchronized final TopologyBuilder setApplicationId(String applicationId) {
```

Comment:
This is also inconsistent with trunk. What are we going to be documenting re: API? I realize this is (presumably) still unstable even though it's not marked as such, but it's going to be confusing if we introduce the same API in both branches but they don't share the same return value...